cli/azd/ci-build.ps1

```diff
@@ -39,6 +39,9 @@ if ($IsWindows) {
     Write-Host "go generate succeeded"
 }
 
+# Force DNS resolution to not use cgo
+$env:GODEBUG='netdns=go+1'
```

Comment:
From the comments in https://go.googlesource.com/go/+/go1.9beta1/src/net/net.go I see that this `+1` here means that the resolve will print debugging information about resolution.  Do we want to remove `+1` now before merging?

Do you understand why these changes ended up being necessary?  I would have expected that on *nix systems we'd be building without cgo to begin with (since we explicitly enable it for the MacOS build, but not here) and reading the go net package documentation it seems that we should have always been using the native resolver.

The 1.20 release notes also made me think that CGO should be enabled less often, not more often, now, so I'm trying to understand why we ended up hitting this problem during our upgrade.

Reply:
@ellismg I meant to convert the PR to draft while testing this out.

The full story is:
1. `net` has always offered a `cgo` and pure-go DNS resolver from their [docs](https://pkg.go.dev/net#hdr-Name_Resolution). `cgo`-based resolver kicks in on OSX, and in some unix distributions.
2. There was a fix to DNS resolution on unix systems. Unfortunately, this [change](https://github.com/golang/go/commit/6939659a085de15ba7e08ebe0c8864616ba21b76) also increased `cgo` surface, by creating a new dependency on glibc `res_search` for DNS resolution. It seems that `golang` isn't interested in removing the `cgo` dependency, [citing ](https://github.com/golang/go/issues/25670#issuecomment-394482195) `nsswitch.conf`'s dynamic loading of C modules as the main point.
3. There is an API compatibility issue between `musl` (on alipne) and `glibc` currently. The symbol `__res_search` was renamed to `res_search` in later `glibc` versions. `musl` has always offered `res_search`.

I was just curious what happens if we force pure-go resolver.  Alternatively, we don't make a change here, simply need to offer `alpine` specific pre-compiled binaries or offer source compilation as an option. It's most likely the latter, but we'll need to discuss this alongside with @danieljurek

Reply:
We're not releasing a compiled binary for musl today... I'm not aware of anyone requesting that support at this time either.

My interest in reducing dependence on `cgo` at this point is to keep compilation as simple as possible for everyone involved. If we do end up using `cgo` then that makes compilation more difficult for affected scenarios because it requires having a correctly configured toolchain on the box and makes cross-compile more difficult.

Reply:
@danieljurek The `cgo` dependency is coming from the `golang` standard library, and it's definitely used in-production on `*nix` machines. Our hands are tied here. Disabling it, to my knowledge, breaks `macOS` and breaks somewhat obscure but complex DNS configurations on unix.

Some questions:

1. Do we need to officially support `alpine`?
2. If so, the two options that enter my mind:
   - Recompiling against `musl`
   - Submitting a change or create a fork that recreates the compatibility layer (high effort)